### ISSUE TITLE
Develop API

### ISSUE THREAD

**Issue by robons** [GSS-Cogs/linked-data-repository#1]
- [ ] Endpoint to return a list of all RDF graphs uploaded to the repos (perhaps searchable?)
- [ ] Endpoint to return a list of the historic versions uploaded for a particular RDF graph
- [ ] Endpoint to upload a versioned CSV-W to the repository (An RDF version of this should be automatically generated on upload).
- [ ] Endpoint to upload a versioned RDF file to the repository.
- [ ] Endpoint to download the RDF (in ttl format) for a given version (`0.1`) of a graph (`some-graph-identifier`)
  - `http://some-base-url/graphs/some-graph-identifier/0.1` with no Accept header OR `Accept: text/plain` (until we develop a suitable human-friendly web page) OR `Accept: text/ttl`
  - `http://some-base-url/graphs/some-graph-identifier/0.1/ttl` with any `Accept` header

**Comment by canwaf** [comment]
Do we want to automatically generate a CSV-W from a given RDF? (Though clearly not something we're building now.)

**Comment by robons** [comment]
> Do we want to automatically generate a CSV-W from a given RDF? (Though clearly not something we're building now.)

I don't think so. There's no way to know we'd be making a sensible CSV-W. Which records do we decide are the ones from the RDF which should be present in the CSV and which ones need to go in the CSV-W? How do you 'correctly' decide?
